source/mesh_deformation/external_tool_interface.cc

```diff
@@ -0,0 +1,518 @@
+/*
+  Copyright (C) 2014 - 2024 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/mesh_deformation/external_tool_interface.h>
+#include <aspect/simulator_signals.h>
+
+#include <deal.II/numerics/vector_tools_evaluate.h>
+
+/*
+TODO:
+- PointDataOut: move to deal.II
+- initial topography from external tool: for now compute_initial_deformation_on_boundary(), later refactor
+  to provide FE vector
+*/
+
+namespace aspect
+{
+  namespace MeshDeformation
+  {
+
+
+
+    template <int dim, int spacedim = dim>
+    class PointDataOut : public dealii::DataOutInterface<0, spacedim>
+    {
+      public:
+        /**
+         * Default constructor.
+         */
+        PointDataOut() = default;
+
+        /**
+         * Default destructor.
+         */
+        ~PointDataOut() = default;
+
+
+        /**
+         * Build the patches for a given set of points and optionally data in each point.
+         *
+         * @param [in] locations The point locations.
+         * @param [in] data A vector of data values for each point.
+         * @param [in] data_component_names An optional vector of strings that
+         * describe the properties of each datum.
+         * @param [in] data_component_interpretations An optional vector that
+         * controls if the properties are interpreted as scalars, vectors,
+         * or tensors. Has to be of the same length as @p data_component_names.
+         */
+        void
+        build_patches(const std::vector<Point<spacedim>> &locations,
+                      const std::vector<std::vector<double>> &data = {},
+                      const std::vector<std::string> &data_component_names = {},
+                      const std::vector<
+                      DataComponentInterpretation::DataComponentInterpretation>
+                      &data_component_interpretations_ = {})
+        {
+          Assert(
+            data_component_names.size() == data_component_interpretations_.size(),
+            ExcMessage(
+              "When calling Particles::DataOut::build_patches with data component "
+              "names and interpretations you need to provide as many data component "
+              "names as interpretations. Provide the same name for components that "
+              "belong to a single vector or tensor."));
```

Comment:
Let's add another couple assertions:
```suggestion
          Assert(
            data_component_names.size() == data_component_interpretations_.size(),
            ExcMessage(
              "When calling Particles::DataOut::build_patches with data component "
              "names and interpretations you need to provide as many data component "
              "names as interpretations. Provide the same name for components that "
              "belong to a single vector or tensor."));
      Assert (data.size() == 0 || data.size() == locations.size(), ...);
      for (const auto &datum : data)
       Assert (datum.size() == data_component_names.size(), ...);
```